backend/httpclient/datasource_metrics_middleware.go

```diff
@@ -0,0 +1,153 @@
+package httpclient
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const kb = 1024
+const mb = kb * kb
+const gb = mb * kb
+
+var (
+	datasourceRequestCounter = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "plugins",
+			Name:      "datasource_request_total",
+			Help:      "A counter for outgoing requests for an external data source",
+		},
+		[]string{"datasource", "datasource_type", "code", "method", "secure_socks_ds_proxy_enabled"},
+	)
+
+	datasourceRequestHistogram = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "plugins",
+			Name:      "datasource_request_duration_seconds",
+			Help:      "histogram of durations of outgoing external data source requests sent from Grafana",
+			Buckets:   []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 25, 50, 100},
```

Comment:
Might want to consider native histogram here as well if it works for `datasource_response_size_bytes`.

Reply:
![image](https://github.com/user-attachments/assets/f24bc452-74a8-40fe-9a06-41ad3d0f6863)
This doesn't look like a native histogram response :/

Reply:
Nope 😞 definitely shows native histograms aren't supported atm

Reply:
I created an issue to investigate this further https://github.com/grafana/grafana-plugin-sdk-go/issues/1132

FYI @wbrowne